analysis_driver/dataset.py

```diff
@@ -403,6 +433,31 @@ def number_of_samples(self):
                 raise AnalysisDriverError('Could not find number of quoted samples in LIMS for ' + self.name)
         return self._number_of_samples
 
+    @property
```

Comment:
Will asserting that only 1 species in the project break some pipelines? e.g. chicken/pheasant and multi species guppies?